hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/HoodieTableSource.java

```diff
@@ -679,12 +678,21 @@ public void reset() {
    * Get the reader paths with partition path expanded.
    */
   @VisibleForTesting
-  public List<StoragePathInfo> getReadFiles() {
+  public List<FileSlice> getBaseFileOnlyFileSlices() {
     List<String> relPartitionPaths = getReadPartitions();
     if (relPartitionPaths.isEmpty()) {
       return Collections.emptyList();
     }
-    return fileIndex.getFilesInPartitions();
+    List<StoragePathInfo> pathInfoList = fileIndex.getFilesInPartitions();
+    try (HoodieTableFileSystemView fsView = new HoodieTableFileSystemView(metaClient,
+        metaClient.getCommitsAndCompactionTimeline().filterCompletedInstants(), pathInfoList)) {
```

Comment:
`metaClient.getCommitsAndCompactionTimeline().filterCompletedAndCompactionInstants()`, let's keep the timeline and the filtering consistent.